src/codegen.cpp

```diff
@@ -544,7 +544,7 @@ static AttributeList get_attrs_ipoeffects(LLVMContext &context, std::pair<bool,u
     if (noinbounds == 1) {
     }
     if (is_decl && consistent == 0 && effect_free == 0 && nothrow == 1 && terminates == 1){
-        attr.addAttribute(Attribute::Speculatable);
```

Comment:
Maybe because we add `nonnull` only on the callee, not the call site?

Reply:
I could try, not sure if that changes something.

Reply:
Even with the nonnulls it still does something weird.